### PR TITLE
Feat/engine to resolve #266

### LINKED ISSUE #266: Add option to error on grule engine Execute
**Is your feature request related to a problem? Please describe.**

The current behaviour of the grule `Execute` function is to ignore rules which access fields that does not exist.

See [Here](https://github.com/hyperjumptech/grule-rule-engine/blob/849aeae2a4b6f56c4ed990fb23561ce6ad168832/engine/GruleEngine.go#L129):

```go
can, err := v.Evaluate(dataCtx, knowledge.WorkingMemory)
if err != nil {
	log.Errorf("Failed testing condition for rule : %s. Got error %v", v.RuleName, err)
	// No longer return error, since unavailability of variable or fact in context might be intentional.
}
```
For my use case, I want to know when rules try to access fields that does not exist.

**Describe the solution you'd like**

To add functionality to report on failed evaluations, a variable can be added to the Grule engine struct, indicating whether an error should be returned in this case. By default the behaviour would be the same, so backwards compatibility would be preserved.

Example struct:

```go
type GruleEngine struct {
	MaxCycle        uint64
	ErrorOnEvaluate bool // New field
	Listeners       []GruleEngineListener
}
```

and an update to Execute:

```go
can, err := v.Evaluate(dataCtx, knowledge.WorkingMemory)
if err != nil {
	log.Errorf("Failed testing condition for rule : %s. Got error %v", v.RuleName, err)
	if g.ErrorOnEvaluate {
	    return err
	}
}
```

After updating the engine can be used as such:

```go
e := engine.NewGruleEngine()
e.ErrorOnEvaluate = true
err := e.Execute(kl, dctx) // Err is populated if path access is attempted that does not exist
```

**Describe alternatives you've considered**

None

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by hjblom**
* Add option to error on failed grule engine execution
 * Add unit test cases

### CODE REVIEW

File: engine/GruleEngine.go [hunk 1]
```diff
@@ -42,8 +42,9 @@ func NewGruleEngine() *GruleEngine {
 
 // GruleEngine is the engine structure. It has the Execute method to start the engine to work.
 type GruleEngine struct {
-	MaxCycle  uint64
-	Listeners []GruleEngineListener
+	MaxCycle             uint64
+	ErrorOnFailedExecute bool
```
Comment (newm4n) [inline engine/GruleEngine.go:46]: Could you use the name like `ReturnErrOnFailedRuleEvaluation`, just to make the variable name become a bit more self explanatory.

**Review by newm4n — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution.
Could you change the variable name so it become more self explanatory.

Again, thanks.

**Comment by hjblom** [comment]
> Thanks a lot for the contribution. Could you change the variable name so it become more self explanatory.
> 
> Again, thanks.

Thanks for the quick review!

**Review by newm4n — APPROVED** [review]
Lets get this merge

**Pull request merged**